code_generation/code_generation.py

```diff
@@ -13,10 +13,10 @@ def fill_template(t, config):
     commandlist = ""  # string to be placed into code template
     df_count = 0  # enumerate dataframes
     df_scope_count = 0  # enumerate dataframes in certain scopes
-
     # get commands of producers and append to the command list
     log.info("Generating commands ...")
     for producer in config["producers"]["global"]:
+        producer.check_output()
```

Comment:
I'd prefer a name like 'reserve_output' to make clear that it changes something and does not only do a look up.

Reply:
the scope will be necessary as argument